apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/EarlySecurityChecks/index.tsx

```diff
@@ -0,0 +1,290 @@
+      device,
+      deviceModelId: deviceModelId,
+      setFirmwareUpdateOpened: () => null, // we don't need to keep the state
+      setFirmwareUpdateCompleted: () => null,
+
+      finalStepSuccessDescription: t(
+        "syncOnboarding.manual.softwareCheckContent.firmwareUpdate.finalStepSuccessDescription",
+      ),
+      finalStepSuccessButtonLabel: t(
+        "syncOnboarding.manual.softwareCheckContent.firmwareUpdate.finalStepSuccessButtonLabel",
+      ),
+      finalStepSuccessButtonOnClick: () => {
+        closeFwUpdateDrawer();
+        restartChecksAfterUpdate();
+      },
+      deviceHasPin: false, // early security checks are triggered only if the device is in one of the steps prior to setting a PIN code
+    };
+
+    setDrawer(UpdateFirmwareModal, updateFirmwareModalProps, {
+      preventBackdropClick: true,
+      forceDisableFocusTrap: true,
+      onRequestClose: () => {
+        closeFwUpdateDrawer();
+        setUpdateInterrupted(true);
+      },
+    });
+  }, [
+    closeFwUpdateDrawer,
+    device,
+    deviceInfo,
+    deviceModelId,
+    latestFirmware,
+    restartChecksAfterUpdate,
+    t,
+  ]);
+
+  useEffect(() => {
+    if (devicePermissionState === "refused") {
+      setGenuineCheckStatus(SoftwareCheckStatus.cancelled);
+    }
+
+    if (genuineState === "non-genuine") {
+      setGenuineCheckStatus(SoftwareCheckStatus.notGenuine);
+    }
+
+    if (genuineState === "genuine") {
+      setGenuineCheckStatus(SoftwareCheckStatus.completed);
+    }
+
+    if (genuineCheckStatus === SoftwareCheckStatus.completed) {
+      setFirmwareUpdateStatus(SoftwareCheckStatus.active);
+    }
+  }, [genuineCheckStatus, genuineState, devicePermissionState, genuineCheckError]);
+
+  useEffect(() => {
+    /**
+     * for infinite loop firmware, we arbitrarily decide that the il2 -> il0
+     * path does not exist. It allows testing this flow without being stuck
+     * in an infinite loop (UX wise)
+     * */
+    const isIL2firmware = deviceInfo?.version.endsWith("-il2");
+    const isIL0firmwareAvailable = latestFirmware?.final?.name?.endsWith("-il0");
+    const shouldPretendNoAvailableFirmware = isIL2firmware && isIL0firmwareAvailable;
+
+    if (firmwareUpdateStatus !== SoftwareCheckStatus.active) return;
+
+    if (
+      (shouldPretendNoAvailableFirmware && status === "available-firmware") ||
+      status === "no-available-firmware"
+    ) {
+      setFirmwareUpdateStatus(SoftwareCheckStatus.completed);
+      setAvailableFirmwareVersion("");
+    } else if (status === "available-firmware") {
+      setAvailableFirmwareVersion(latestFirmware?.final.name || "");
+      setFirmwareUpdateStatus(SoftwareCheckStatus.updateAvailable);
+    }
+  }, [firmwareUpdateStatus, status, genuineCheckStatus, latestFirmware, deviceInfo?.version]);
+
+  const lockedDeviceModalIsOpen =
+    (devicePermissionState === "unlock-needed" && genuineCheckActive) ||
+    (lockedDevice && firmwareUpdateStatus === SoftwareCheckStatus.active);
```

Comment:
You should && another condition: 

```TS
        // The device has no PIN in the ESC - and having an unresponsive device here probably means the user
        // cancelled a firmware update previously and came back to the ESC (with a device in an incorrect state).
        // -> Not displaying a drawer in the case of unresponsive, and letting the device coming back to a normal state.
        (lockedDevice && firmwareUpdateStatus === SoftwareCheckStatus.active && getLatestAvailableFirmwareError?.name !== "UnresponsiveDeviceError") {
```

Reply:
I mean: i'm thinking right now on LLM: maybe we should only log that the device was considered locked, but not display anything ? Because for the early security check, locking a device should be impossible (except if it's considered "unresponsive" like i was saying above, but it's different than locked in this specific case)

Reply:
And we should move this locked device drawer inside the screen where the polling occurs (to know if the ESC or the sync companion should be rendered): in this case it could happen that the device is locked, and you can check on the `allowedError`

Reply:
Actually, i already return an `lockedDevice` from `useOnboardingStatePolling` ;)